src/openloco/windows/CompanyWindow.cpp

```diff
@@ -847,10 +847,49 @@ namespace openloco::ui::windows::CompanyWindow
             // Set company's secondary colour
             self->widgets[widx::secondary_colour_scheme].image = (1 << 30) | gfx::recolour(image_ids::colour_swatch_recolourable, company->mainColours.secondary);
 
-            // TODO: set colours for different ride types.
-            registers regs;
-            regs.esi = (int32_t)self;
-            call(0x00432EB9, regs);
+            struct ColourSchemeTuple
+            {
+                widget_index checkbox;
+                widget_index primary;
+                widget_index secondary;
+            };
+
+            // clang-format off
+            static const ColourSchemeTuple tuples[] =
+            {
+                { widx::check_steam_locomotives,     widx::main_colour_steam_locomotives,     widx::secondary_colour_steam_locomotives },
+                { widx::check_diesel_locomotives,    widx::main_colour_diesel_locomotives,    widx::secondary_colour_diesel_locomotives },
+                { widx::check_electric_locomotives,  widx::main_colour_electric_locomotives,  widx::secondary_colour_electric_locomotives },
+                { widx::check_multiple_units,        widx::main_colour_multiple_units,        widx::secondary_colour_multiple_units },
+                { widx::check_passenger_vehicles,    widx::main_colour_passenger_vehicles,    widx::secondary_colour_passenger_vehicles },
+                { widx::check_freight_vehicles,      widx::main_colour_freight_vehicles,      widx::secondary_colour_freight_vehicles },
+                { widx::check_buses,                 widx::main_colour_buses,                 widx::secondary_colour_buses },
+                { widx::check_trucks,                widx::main_colour_trucks,                widx::secondary_colour_trucks },
+                { widx::check_aircraft,              widx::main_colour_aircraft,              widx::secondary_colour_aircraft },
+                { widx::check_ships,                 widx::main_colour_ships,                 widx::secondary_colour_ships },
+            };
+            // clang-format on
+
+            for (uint8_t i = 0; i < static_cast<uint8_t>(std::size(tuples)); i++)
+            {
+                if ((company->customVehicleColoursSet & (1 << i)) != 0)
```

Comment:
This appears to be off by one.

master:
![Screenshot](https://user-images.githubusercontent.com/604665/80307285-57889800-87c8-11ea-8ea8-952d8abf34d2.png)

this implementation:
![Screenshot (1)](https://user-images.githubusercontent.com/604665/80307284-56f00180-87c8-11ea-9754-930f3b147067.png)

Reply:
The first bit appears to be used for something else — it's skipped in the checks. I've added a new commit to compensate for this.

Reply:
The first bit is for the main colour scheme even though it is not used. It makes it easier for using the same offsets when used with the colour array.